src/express.re

```diff
@@ -1,39 +1,147 @@
+
+  external hostname : t => string = "" [@@bs.get];
+  /** [hostname request] Contains the hostname derived from the Host
+      HTTP header.*/
+
+  external ip : t => string = "" [@@bs.get];
+  /** [ip request] Contains the remote IP address of the request.*/
+
+  external fresh : t => bool =  "" [@@bs.get];
+  /** [fresh request] returns [true] whether the request is "fresh" */
+
+  external stale : t => bool =  "" [@@bs.get];
+  /** [stale request] returns [true] whether the request is "stale"*/
+
+  external methodRaw : t => string = "method" [@@bs.get];
+
+  type method_ =
+    | Get
+    | Post
+    | Put
+    | Delete
+    | Head
+    | Options
+    | Trace
+    | Connect;
+
+  let method_ : t => method_ = fun req => {
+    switch (methodRaw req) {
+      | "GET" => Get;
+      | "POST" => Post;
+      | "PUT" => Put;
+      | "DELETE" => Delete;
+      | "HEAD" => Head;
+      | "OPTIONS" => Options;
+      | "TRACE" => Trace;
+      | "CONNECT" => Connect;
+      | s => failwith ("Express.Request.method_ Unexpected method: " ^ s);
+    }};
+  /** [method_ request] return a string corresponding to the HTTP
+      method of the request: GET, POST, PUT, and so on */
+
+  external originalUrl : t => string = "" [@@bs.get];
+  /** [originalUrl request] returns the original url. See
+      https://expressjs.com/en/4x/api.html#req.originalUrl */
+
+  external path : t => string = "" [@@bs.get];
+  /** [path request] returns the path part of the request URL.*/
+
+  external protocolRaw : t => string = "protocol" [@@bs.get];
+
+  type protocol =
+    | Http
+    | Https;
+
+  let protocol : t => protocol = fun req => {
+    switch (protocolRaw req) {
+      | "http" => Http;
+      | "https" => Https;
+      | s => failwith ("Express.Request.protocol, Unexpected protocol: " ^ s)
+    }};
+  /** [protocol request] returns the request protocol string: either http
+      or (for TLS requests) https. */
+
+  external secure : t => bool = "" [@@bs.get];
+  /** [secure request] returns [true] if a TLS connection is established */
+
+  external query : t => Js.Dict.t Js.Json.t = "" [@@bs.get];
+  /** [query request] returns an object containing a property for each
+      query string parameter in the route. If there is no query string,
+      it returns the empty object, {} */
+
+  external acceptsRaw : t => array string => Js.Json.t = "accepts" [@@bs.send];
+  /** [acceptsRaw request types] checks if the specified content types
+      are acceptable, based on the request's Accept HTTP header field.
+      The method returns the best match, or if none of the specified
+      content types is acceptable, returns [false] */
+
+  let accepts : t => array string => option string = fun req types => {
+    let ret = acceptsRaw req types;
+    let (ty, x) = Js.Json.reifyType ret;
+    switch ty {
+    | Js_json.String => Some x
+    | _ => None
+    };
+  };
+
+  external acceptsCharsetsRaw : t => array string => Js.Json.t
```

Comment:
Also, if they're not supposed to be exposed and until there is an interface file, it would be nice to have non-public functions marked `/** internal */`